### ISSUE TITLE
[SOLVED] `npm start` gives multiple error messages

### ISSUE THREAD

**Issue by richb-hanover** [SpellCraft/mdb4-boilerplate#7]
After `npm install` (see #6), `npm start` gives an error using node 10.15.1 & npm 6.4.1 on macOS 10.12.6. The browser opened with the basic page, but there were CSS problems. In addition, the following error messages appeared.

Thoughts about addressing this? Thanks.

```
bash-3.2$ npm start

> mdb4-boilerplate@2.0.0 start /Users/richb/github/mdb4-boilerplate
> webpack-dev-server --mode development --progress --colors --hot --open --inline --compress

clean-webpack-plugin: /Users/richb/github/mdb4-boilerplate/dist has been removed.
 10% building 1/1 modules 0 activeℹ ｢wds｣: Project is running at http://localhost:8080/
ℹ ｢wds｣: webpack output is served from /
 10% building 5/13 modules 8 active ...modules/webpack/hot/log-apply-result.jsℹ ｢wdm｣: wait until bundle finished: /
✖ ｢wdm｣: Hash: 597591c445c2e03492db
Version: webpack 4.28.3
Time: 2182ms
Built at: 05/05/2019 7:58:20 PM
                    Asset      Size        Chunks                    Chunk Names
               index.html  1.22 KiB                [emitted]
             js/bundle.js  39.1 KiB          main  [emitted]         main
js/vendors~main.bundle.js  2.01 MiB  vendors~main  [emitted]  [big]  vendors~main
Entrypoint main [big] = js/vendors~main.bundle.js js/bundle.js
[0] multi (webpack)-dev-server/client?http://localhost:8080 (webpack)/hot/dev-server.js ./src/js/app.js ./src/scss/main.scss 64 bytes {main} [built]
[./node_modules/@fortawesome/fontawesome-free/js/all.js] 1.12 MiB {vendors~main} [built]
[./node_modules/bootstrap/dist/js/bootstrap.js] 123 KiB {vendors~main} [built]
[./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/lib/loader.js!./src/scss/main.scss] 297 bytes {main} [built] [failed] [1 error]
[./node_modules/jquery/dist/jquery.js] 274 KiB {vendors~main} [built]
[./node_modules/loglevel/lib/loglevel.js] 7.68 KiB {vendors~main} [built]
[./node_modules/popper.js/dist/esm/popper.js] 84.7 KiB {vendors~main} [built]
[./node_modules/style-loader/lib/addStyles.js] 10.2 KiB {vendors~main} [built]
[./node_modules/url/url.js] 22.8 KiB {vendors~main} [built]
[./node_modules/webpack-dev-server/client/index.js?http://localhost:8080] (webpack)-dev-server/client?http://localhost:8080 7.78 KiB {vendors~main} [built]
[./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.58 KiB {vendors~main} [built]
[./node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 1.05 KiB {vendors~main} [built]
[./node_modules/webpack/hot/dev-server.js] (webpack)/hot/dev-server.js 1.61 KiB {vendors~main} [built]
[./src/js/app.js] 536 bytes {main} [built] [1 error]
[./src/scss/main.scss] 1.2 KiB {main} [built]
    + 23 hidden modules

ERROR in ./src/js/app.js
Module Error (from ./node_modules/eslint-loader/index.js):

/Users/richb/github/mdb4-boilerplate/src/js/app.js
  30:8  error  Unable to resolve path to module '../vendors/mdb/js/mdb'                import/no-unresolved
  31:8  error  Unable to resolve path to module '../vendors/mdb/js/addons/datatables'  import/no-unresolved

✖ 2 problems (2 errors, 0 warnings)

 @ multi (webpack)-dev-server/client?http://localhost:8080 (webpack)/hot/dev-server.js ./src/js/app.js ./src/scss/main.scss main[2]

ERROR in ./src/scss/main.scss (./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/lib/loader.js!./src/scss/main.scss)
Module build failed (from ./node_modules/sass-loader/lib/loader.js):

@import '../vendors/mdb/scss/mdb.scss';
^
      File to import not found or unreadable: ../vendors/mdb/scss/mdb.scss.
      in /Users/richb/github/mdb4-boilerplate/src/scss/main.scss (line 19, column 1)
 @ ./src/scss/main.scss 2:14-125 21:1-42:3 22:19-130
 @ multi (webpack)-dev-server/client?http://localhost:8080 (webpack)/hot/dev-server.js ./src/js/app.js ./src/scss/main.scss

ERROR in ./src/js/app.js
Module not found: Error: Can't resolve '../vendors/mdb/js/addons/datatables' in '/Users/richb/github/mdb4-boilerplate/src/js'
 @ ./src/js/app.js 32:0-45
 @ multi (webpack)-dev-server/client?http://localhost:8080 (webpack)/hot/dev-server.js ./src/js/app.js ./src/scss/main.scss

ERROR in ./src/js/app.js
Module not found: Error: Can't resolve '../vendors/mdb/js/mdb' in '/Users/richb/github/mdb4-boilerplate/src/js'
 @ ./src/js/app.js 31:0-31
 @ multi (webpack)-dev-server/client?http://localhost:8080 (webpack)/hot/dev-server.js ./src/js/app.js ./src/scss/main.scss
Child html-webpack-plugin for "index.html":
     1 asset
    Entrypoint undefined = index.html
    [./node_modules/html-webpack-plugin/lib/loader.js!./src/index.html] 1.14 KiB {0} [built]
ℹ ｢wdm｣: Failed to compile.
```

**Comment by SpellCraft** [comment]
Thanks for the report!

I've been working on updating the repository this weekend to support the newest version of Webpack and Babel, along with MDBootstrap 4.8.0 (Pro and Free), however 4.8.0 seems to have issues with Webpack 4 currently because of the way they import their vendors and export the project.

I'll publish an update based on MDB v4.7.7 in a couple of hours once I've run some tests.

**Comment by richb-hanover** [comment]
I got it working - the cause of the error was that I did not use `git clone --recursive` when I initially created the repository. (I used a GUI git client, and wasn't aware that there was a recursive option.)

Once the vendors/mdb files were actually present, the sample project worked as expected. Thanks.

**Comment by SpellCraft** [comment]
Ah! Yeah. Originally, I wanted MDB to automatically install together with the project when checked out, this is fortunately fixed in the upcoming update.

I figured out how to fix 4.8.0 to work with NPM so expect an update that works out-of-the-box sometime tomorrow without the need to clone the project recursively. :)
